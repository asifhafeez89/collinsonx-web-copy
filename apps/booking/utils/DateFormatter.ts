import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import { DATE_TIME_FORMAT } from 'config/Constants';
import 'dayjs/locale/es';
dayjs.extend(utc);
dayjs.extend(timezone);

export function formatDate(
  dateval: Date,
  formattype: string,
  locale?: string
): string {
  dayjs.locale(locale ? locale : 'en');
  return dayjs(dateval).format(formattype);
}

export function formatTimezone(localTimeHour: string, timezone: string) {
  return dayjs.tz(localTimeHour, timezone).format(DATE_TIME_FORMAT);
}
