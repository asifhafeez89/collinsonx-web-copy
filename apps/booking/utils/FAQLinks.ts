export const FAQLink = (accountProvider?: string, locale?: string) => {
  const loungeKey_link_faq = `https://portal.loungekey.com/${locale}/elo/faq/`;
  const priorityPass_link_faq = `https://memberhelp.prioritypass.com/${locale}/support/home`;
  return accountProvider === 'LOUNGE_KEY'
    ? loungeKey_link_faq
    : accountProvider === 'PRIORITY_PASS'
    ? priorityPass_link_faq
    : '';
};
