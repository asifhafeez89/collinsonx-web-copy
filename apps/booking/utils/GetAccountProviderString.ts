export const GetAccountProviderString = (accountProvider?: string) => {
  switch (accountProvider) {
    case 'LOUNGE_KEY':
      return 'Lounge Key';
    case 'PRIORITY_PASS':
      return 'Priority Pass';
    default:
      return 'Priority Pass';
  }
};
