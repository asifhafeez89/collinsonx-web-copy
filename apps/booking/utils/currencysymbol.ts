export const currencyMap: Record<string, string> = {
  GBP: String.fromCharCode(163),
};

export const getCurrencySymbol = (currency: string) =>
  currencyMap[currency] || currency;
