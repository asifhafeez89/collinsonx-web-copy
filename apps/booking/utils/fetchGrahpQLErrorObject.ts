// @ts-ignore
function fetchGrahpQLErrorObject(response, key: string) {
  if (!response && typeof response !== 'object') {
    return null;
  }

  if (!response[key]) {
    return null;
  }

  const error = response[key];

  if (error && 'graphQLErrors' in error) {
    const graphQLErrors = error.graphQLErrors;

    if (graphQLErrors && typeof graphQLErrors === 'object') {
      const data = error.graphQLErrors[0];
      if ('extensions' in data) {
        return data.extensions;
      }
    }
  }

  return null;
}

export default fetchGrahpQLErrorObject;
