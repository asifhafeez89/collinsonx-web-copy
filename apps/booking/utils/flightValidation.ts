const FLIGHT_VALIDATION = /([0-9]{0,1}[a-zA-Z]{1,3})([0-9]*)/gi;

export function validateFlightNumber(
  flight: string
): [boolean, string | undefined, string | undefined] {
  const match = [...flight.matchAll(FLIGHT_VALIDATION)];

  if (!match) {
    return [false, undefined, undefined];
  }

  const carrierLen = match[0]?.[1]?.length ?? 0;
  if (carrierLen < 1 || carrierLen > 3) {
    return [false, undefined, undefined];
  }

  const flightNoLen = match[0]?.[2]?.length ?? 0;
  if (flightNoLen < 1 || flightNoLen > 5) {
    return [false, undefined, undefined];
  }

  return [true, match[0][1], match[0][2]];
}
