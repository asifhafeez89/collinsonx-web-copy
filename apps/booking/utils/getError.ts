import { isMapIterator } from 'util/types';
import { BookingError } from '../constants';

export default function getError(response: any, errorCode: BookingError) {
  if (response && response.errors) {
    return response.errors.find(
      (item: any) => item.extensions.code === errorCode
    );
  }
}
