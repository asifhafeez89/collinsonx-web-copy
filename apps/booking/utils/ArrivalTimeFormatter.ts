import { formatDate } from './DateFormatter';
import { TIME_FORMAT } from 'config/Constants';

export const arrivalTimeFormatter = (
  bookedFrom: string | undefined,
  lastArrival: string | undefined
) =>
  `${formatDate(new Date(`${bookedFrom}`), TIME_FORMAT)}
  -
  ${formatDate(new Date(`${lastArrival}`), TIME_FORMAT)}`;
