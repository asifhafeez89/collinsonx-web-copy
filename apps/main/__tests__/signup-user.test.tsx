import * as React from 'react';
import renderer from 'react-test-renderer';
import SignupUser from '../pages/signup-user';
import { MockedProvider } from '@collinsonx/utils/testing';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

describe('<SignupUser />', () => {
  it('renders outline variant', () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[]} addTypename={false}>
          <SignupUser />
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
