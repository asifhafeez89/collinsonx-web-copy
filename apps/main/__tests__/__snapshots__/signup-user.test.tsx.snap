// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupUser /> renders outline variant 1`] = `
<div
  className="mantine-Container-root mantine-hwrsz4"
>
  <div
    className="mantine-Stack-root mantine-g49psp"
  >
    <div
      className="mantine-Center-root mantine-dejthu"
    >
      svg
       
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="mantine-Stack-root mantine-1wpa508"
      >
        <div
          className="mantine-Stack-root mantine-shhoda"
        >
          <header
            className="mantine-1loib0s"
          >
            <div
              className="mantine-jrn8go"
              onClick={[Function]}
            >
              <SvgrURL
                color="#FFFFFF"
              />
            </div>
            <h1
              className="mantine-Text-root mantine-Title-root mantine-xichux"
            >
              Sign up with email
            </h1>
          </header>
          <div
            className="mantine-InputWrapper-root mantine-TextInput-root mantine-agt6mi"
          >
            <label
              className="mantine-InputWrapper-label mantine-TextInput-label mantine-1xp1947"
              htmlFor="mantine-rf0jhgruc"
              id="mantine-rf0jhgruc-label"
              onMouseDown={[Function]}
            >
              First name(s)
              <span
                aria-hidden={true}
                className="mantine-103svbs mantine-InputWrapper-required mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-7c7vou"
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="mantine-Input-input mantine-TextInput-input mantine-1o8gqm4"
                id="mantine-rf0jhgruc"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="First name"
                type="text"
              />
            </div>
          </div>
          <div
            className="mantine-InputWrapper-root mantine-TextInput-root mantine-agt6mi"
          >
            <label
              className="mantine-InputWrapper-label mantine-TextInput-label mantine-1xp1947"
              htmlFor="mantine-1c343tpo7"
              id="mantine-1c343tpo7-label"
              onMouseDown={[Function]}
            >
              Last name
              <span
                aria-hidden={true}
                className="mantine-103svbs mantine-InputWrapper-required mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-7c7vou"
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="mantine-Input-input mantine-TextInput-input mantine-1o8gqm4"
                id="mantine-1c343tpo7"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Last name"
                type="text"
              />
            </div>
          </div>
          <div
            className="mantine-InputWrapper-root mantine-DatePickerInput-root mantine-1xhjld9"
          >
            <label
              className="mantine-InputWrapper-label mantine-DatePickerInput-label mantine-1fzet7j"
              htmlFor="mantine-955dojawf"
              id="mantine-955dojawf-label"
              onMouseDown={[Function]}
            >
              Date of birth
            </label>
            <div
              className="mantine-Input-wrapper mantine-DatePickerInput-wrapper mantine-7c7vou"
            >
              <div
                className="mantine-16k3d8y mantine-Input-icon mantine-DatePickerInput-icon"
              >
                icon
              </div>
              <button
                aria-controls="mantine-jxo7wb92e-dropdown"
                aria-expanded={false}
                aria-haspopup="dialog"
                aria-invalid={false}
                className="mantine-DatePickerInput-input mantine-Input-input mantine-DatePickerInput-input mantine-1jbpix9"
                data-dates-input={true}
                data-with-icon={true}
                id="mantine-jxo7wb92e-target"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                type="button"
              >
                01/01/1990
              </button>
            </div>
          </div>
          <input
            type="hidden"
            value="1990-01-01T00:00:00.000Z"
          />
          <div
            className="mantine-InputWrapper-root mantine-TextInput-root mantine-agt6mi"
          >
            <label
              className="mantine-InputWrapper-label mantine-TextInput-label mantine-1xp1947"
              htmlFor="mantine-4a3otjhyc"
              id="mantine-4a3otjhyc-label"
              onMouseDown={[Function]}
            >
              Your email address
              <span
                aria-hidden={true}
                className="mantine-103svbs mantine-InputWrapper-required mantine-TextInput-required"
              >
                 *
              </span>
            </label>
            <div
              className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-7c7vou"
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="mantine-Input-input mantine-TextInput-input mantine-1o8gqm4"
                id="mantine-4a3otjhyc"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Your email address"
                readOnly={true}
                type="email"
              />
            </div>
          </div>
          <div
            className="mantine-Text-root mantine-1d564l0"
          >
            We will send you a unique code via email to complete the login.
          </div>
          <div
            className="mantine-Group-root mantine-k3ov3c"
          >
            <div
              className="mantine-1r6lzvf"
            >
              <div
                className="mantine-Checkbox-root mantine-yxmaw9"
              >
                <div
                  className="mantine-14n9mxy mantine-Checkbox-body"
                >
                  <div
                    className="mantine-1vf457v mantine-Checkbox-inner"
                  >
                    <input
                      checked={false}
                      className="mantine-qjqn4g mantine-Checkbox-input"
                      id="mantine-2kka5v3mt"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="checkbox"
                    />
                    <svg
                      className="___ref-icon mantine-q9qwbl mantine-Checkbox-icon"
                      fill="none"
                      viewBox="0 0 10 7"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clipRule="evenodd"
                        d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z"
                        fill="currentColor"
                        fillRule="evenodd"
                      />
                    </svg>
                  </div>
                  <div
                    className="mantine-1svz3tp mantine-Checkbox-labelWrapper"
                  >
                    <label
                      className="mantine-1y85e1c mantine-Checkbox-label"
                      htmlFor="mantine-2kka5v3mt"
                    >
                      I agree to receive personalised marketing emails.
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="mantine-Group-root mantine-k3ov3c"
          >
            <button
              className="mantine-UnstyledButton-root mantine-Button-root mantine-1pa5pmq"
              data-button={true}
              type="submit"
            >
              <div
                className="mantine-1wpc1xj mantine-Button-inner"
              >
                <span
                  className="mantine-1ryt1ht mantine-Button-label"
                  style={
                    {
                      "textTransform": undefined,
                    }
                  }
                >
                  Login
                </span>
              </div>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
